tests/report/polarion/test.sh

```diff
@@ -7,14 +7,125 @@ rlJournalStart
+        rlAssertGrep '<property name="polarion-project-span-ids" value="myprojectid,RHELBASEOS"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-testrun-title" value="mytitle"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-testrun-template-id" value="mytemplate"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-user-id" value="' "xunit.xml"
+
+        # Custom testsuite properties
+        rlAssertGrep '<property name="polarion-custom-description" value="mydesc"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-custom-plannedin" value="myplannedin"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-custom-assignee" value="myassignee"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-custom-poolteam" value="mypoolteam"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-custom-arch" value="x86_64"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-custom-platform" value="myplatform"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-custom-build" value="mybuild"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-custom-sampleimage" value="mysampleimage"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-custom-logs" value="mylogslocation"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-custom-composeid" value="mycomposeid"/>' "xunit.xml"
+
+        # The testcase properties
+        rlAssertGrep '<property name="polarion-testcase-id" value="BASEOS-10913"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-testcase-id" value="BASEOS-10914"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-testcase-id" value="BASEOS-10915"/>' "xunit.xml"
+        rlAssertGrep '<property name="polarion-testcase-project-id" value="RHELBASEOS"/>' "xunit.xml"
+    rlPhaseEnd
+
+    rlPhaseStartTest 'Check the plugin behavior based on TMT_PLUGIN_REPORT_POLARION_USE_FACTS env variable'
+        # Make sure all ENV variables are unset
+        rlRun "unset \
+        TMT_PLUGIN_REPORT_POLARION_PROJECT_ID \
+        TMT_PLUGIN_REPORT_POLARION_TITLE \
+        TMT_PLUGIN_REPORT_POLARION_DESCRIPTION \
+        TMT_PLUGIN_REPORT_POLARION_TEMPLATE \
+        TMT_PLUGIN_REPORT_POLARION_PLANNED_IN \
+        TMT_PLUGIN_REPORT_POLARION_ASSIGNEE \
+        TMT_PLUGIN_REPORT_POLARION_POOL_TEAM \
+        TMT_PLUGIN_REPORT_POLARION_ARCH \
+        TMT_PLUGIN_REPORT_POLARION_PLATFORM \
+        TMT_PLUGIN_REPORT_POLARION_BUILD \
+        TMT_PLUGIN_REPORT_POLARION_SAMPLE_IMAGE \
+        TMT_PLUGIN_REPORT_POLARION_LOGS \
+        TMT_PLUGIN_REPORT_POLARION_COMPOSE_ID \
+        TMT_PLUGIN_REPORT_POLARION_USE_FACTS \
+        "
+
+        # The facts must not be set
+        rlRun "export TMT_PLUGIN_REPORT_POLARION_USE_FACTS=0"
+        rlRun "tmt run -avr execute report -h polarion --no-upload --file xunit.xml 2>&1 >/dev/null | tee output" 2
```

Comment:
I would expect this to fail since you have no project id set

Reply:
was thinking that none of these can actually run and yeah, these are not run in the pipelines so they can be run locally only...and the project_id would be taken from your pylero config...it's fun returning to own code couple months/years later 😆

Reply:
Yeah, I agree. They are not failing on my machine due to `PolarionWorkItem._session.default_project`. The tests will probably fail in the CI - I will fix them by providing `--project-id` and I will probably add one more test which will check the tmt fails if no `project_id` is found.

Reply:
But firstly I need to resolve the lxml installation problems in the CI :/ (https://github.com/teemtee/tmt/pull/3150#issuecomment-2315937339).

Reply:
thanks, IMO it's better to always have explicit project ID

but as mentioned in the comment CI is not running these, tests/report/polarion/main.fmf clearly states `enabled: false` since we don't have any credentials in there and also no access to a Polarion instance

may be moved into teemtee/tests and test with requre, but am not sure if it is doable, definitely not in scope of this patch

Reply:
> `tests/report/polarion/main.fmf` clearly states `enabled: false`

Oh, I missed that :man_facepalming: . Ok never mind. Let's just for now keep the tests here for local testing? I will add an explicit `--project-id` to these tests.